content/posts/2018-01-10-kotlin-null.md

```diff
@@ -0,0 +1,249 @@
+---
+title: "Kotlin: The Problem with `null`"
+date: 2018-01-10
+slug: kotlin-the-problem-with-null
+---
+
+I think almost everybody in the CS world knows about the [Null References: The Billion Dollar Mistake](https://www.infoq.com/presentations/Null-References-The-Billion-Dollar-Mistake-Tony-Hoare) talk.
+However, it does not really matter if a concept of `null` references a good idea or a really bad one.
+We live in the world where we just have to deal with `null` this way or another.
+Yep, just deal with it :sunglasses:. So let’s ask the “How” question instead of the “Why” one.
+
+# Enter Kotlin
+
+The drug of choice of mine at this point in time is Kotlin,
+but you cannot forget about Java when using Kotlin.
+Well, the JVM probably can be avoided by using Kotlin Native,
+but at least on Android you are stuck with the JVM as an intermediate bytecode provider.
```

Comment:
"with the JVM as an intermediate bytecode provider."

You can add "as well as runtime semantics"